Home Work Solution

dlvsn opened the pull request:
Created classes. Made inheritance. Overrided methods from Abstract Class. Added constructor to each class

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Override
+    public void doWork() {
+        System.out.println(getNameMachine() + " start work");
```
The message should be more informative and indicate that the Bulldozer is starting to work. Consider specifying the type of machine in the message, for example: 'Bulldozer ' + getNameMachine() + ' started working.'

dlvsn replied:
can you apply my hm ?

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Override
+    public void stopWork() {
+        System.out.println(getNameMachine() + " stoped work");
```
There is a typo in the word 'stopped'. Additionally, the stop message should also be more informative. You could rephrase it to 'Bulldozer ' + getNameMachine() + ' stopped working.' to clearly indicate the type of machine and its state.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Override
+    public void doWork() {
+        System.out.println(getNameMachine() + " start work");
```
The message in the 'doWork' method is not informative enough. It should indicate that an 'Excavator' is working. Consider changing the message to include the type of the machine, for example: 'Excavator ' + getNameMachine() + ' started working.'

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Override
+    public void stopWork() {
+        System.out.println(getNameMachine() + " stoped work");
```
Similarly, the message in the 'stopWork' method should be more informative and indicate that an 'Excavator' has stopped working. Also, there is a typo in 'stoped', which should be 'stopped'. Consider changing the message to: 'Excavator ' + getNameMachine() + ' stopped working.'

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/MainApp.java:
```diff
@@ -2,4 +2,16 @@
 
+    public static void main(String[] args) {
+        Machine[] machines = new Machine[] {
+                new Bulldozer("Buldozer"),
+                new Excavator("Excavator"),
+                new Truck("Truck")
```
It seems like you are passing a string parameter to the constructors of `Bulldozer`, `Excavator`, and `Truck`, which is not necessary for the task. Instead, the message should be generated within the `doWork` and `stopWork` methods of each machine type.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/MainApp.java:
```diff
@@ -2,4 +2,16 @@
+                new Truck("Truck")
+        };
+
+        for (int i = 0; i < machines.length; i++) {
+            machines[i].doWork();
+            machines[i].stopWork();
```
While the code inside the loop is correct, consider enhancing the readability of your loop by using the enhanced for-loop (for-each loop). This will make your code cleaner and easier to understand.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Override
+    public void doWork() {
+        System.out.println(getNameMachine() + " started working");
```
The message printed should be more informative. It should indicate that a 'Truck' is the type of machine that started working. For example, 'Truck <name> started working.'

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Override
+    public void stopWork() {
+        System.out.println(getNameMachine() + " stoped working");
```
There is a typo in the message 'stoped'. The correct spelling is 'stopped'. Also, the message should be more informative and indicate that a 'Truck' is the type of machine that stopped working. For example, 'Truck <name> stopped working.'

dlvsn commented:
Edited

dlvsn commented:
Can u apply my home work ?

dlvsn commented:
> Created classes. Made inheritance. Overrided methods from Abstract Class. Added constructor to each class

Removed constructor and made methods more informative

dlvsn reviewed (commented):
fixed the errors that the mentor wrote about. Removed constructors and made methods more informative

Elena-Bruyako commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Bulldozer extends Machine {
+
```
```suggestion
```

Elena-Bruyako commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Excavator extends Machine {
+
```
```suggestion
```
Don't start class or method from empty line

Elena-Bruyako commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
+public abstract class Machine {
+
```
```suggestion
```

Elena-Bruyako commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Truck extends Machine {
+
```
```suggestion
```

Elena-Bruyako commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
```
Please, remove DS_Store files
